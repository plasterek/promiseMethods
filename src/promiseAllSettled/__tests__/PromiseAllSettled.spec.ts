import { promiseAllSettled } from "../PromiseAllSettled.service";

describe("promiseAllSettled function", () => {
  const resolvedPromise: () => Promise<unknown> = () => Promise.resolve("resolved");
  const rejectedPromise: () => Promise<unknown> = () => Promise.reject("rejected");
  describe("When any promise in given array is resolved", () => {
    it("It returns Promise with array of objects with given promise status and results", () => {
      //given
      const promiseArray: (() => Promise<unknown>)[] = [resolvedPromise, resolvedPromise];
      //when
      const results: Promise<unknown> = promiseAllSettled(promiseArray);
      //then
      expect.assertions(1);
      expect(results).resolves.toMatchSnapshot();
    });
  });
  describe("When any promise rejects in given array", () => {
    it("It will be returned in promise array with status and reason", () => {
      //given
      const promiseArray: (() => Promise<unknown>)[] = [resolvedPromise, rejectedPromise];
      //when
      const results: Promise<unknown> = promiseAllSettled(promiseArray);
      //then
      expect.assertions(1);
      expect(results).resolves.toMatchSnapshot();
    });
  });
  describe("When given array is empty", () => {
    it("It returns Promise with an empty array", () => {
      //given
      const promiseArray: any[] = [];
      //when
      const results: Promise<unknown> = promiseAllSettled(promiseArray);
      //then
      expect.assertions(1);
      expect(results).resolves.toMatchSnapshot();
    });
  });
});
