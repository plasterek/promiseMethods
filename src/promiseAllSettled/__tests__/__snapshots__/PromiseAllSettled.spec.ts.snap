// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promiseAllSettled function When any promise in given array is resolved It returns Promise with array of objects with given promise status and results 1`] = `
[
  {
    "status": "fulfilled",
    "value": "resolved",
  },
  {
    "status": "fulfilled",
    "value": "resolved",
  },
]
`;

exports[`promiseAllSettled function When any promise rejects in given array It will be returned in promise array with status and reason 1`] = `
[
  {
    "status": "fulfilled",
    "value": "resolved",
  },
  {
    "reason": "rejected",
    "status": "rejected",
  },
]
`;

exports[`promiseAllSettled function When given array is empty It returns Promise with an empty array 1`] = `[]`;
